pkg/certificate/castorage/k8s/k8s.go

```diff
@@ -43,7 +43,7 @@ func GetCertFromKubernetes(ns string, secretName string, kubeClient kubernetes.I
 		return nil, certificate.ErrInvalidCertSecret
 	}
 
-	cert, err := certificate.NewFromPEM(pemCert, pemKey)
+	cert, err := certificate.NewRootCertificateFromPEM(pemCert, pemKey)
```

Comment:
I'm slightly concerned about the ways that we're using the term "root certificate". By definition, a root certificate is the last node in the trust chain. This isn't at all blocking, but I wonder if we could remove some ambiguity by being stricter with our terms